objects/src/accounts/package/storage_entry/word.rs

```diff
@@ -0,0 +1,266 @@
+                } else {
+                    Err(DeError::invalid_length(elements.len(), &"expected an array of 4 elements"))
+                }
+            }
+        }
+
+        deserializer.deserialize_any(WordRepresentationVisitor)
+    }
+}
+
+impl Default for WordRepresentation {
+    fn default() -> Self {
+        WordRepresentation::SingleHex(Default::default())
+    }
+}
+
+// FELTS
+// ================================================================================================
+
+/// Supported element representations in TOML format. Represents slot values and keys.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum FeltRepresentation {
+    /// Hexadecimal representation of a field element.
+    SingleHex(Felt),
+    /// Single decimal representation of a field element.
+    SingleDecimal(Felt),
+    /// A template key written as "{{key}}".
+    Dynamic(TemplateKey),
+}
+
+impl FeltRepresentation {
+    pub fn template_keys(&self) -> impl Iterator<Item = &TemplateKey> {
+        let maybe_key = match self {
+            FeltRepresentation::Dynamic(template_key) => Some(template_key),
+            _ => None,
+        };
+
+        maybe_key.into_iter()
+    }
+
+    /// Attempts to convert the [FeltRepresentation] into a [Felt].
+    ///
+    /// If the representation is dynamic, the value is retrieved from `template_values`, identified
+    /// by its key. Otherwise, the returned value is just the inner element.
+    pub fn try_into_felt(
+        self,
+        template_values: &BTreeMap<String, TemplateValue>,
+    ) -> Result<Felt, AccountComponentTemplateError> {
+        match self {
+            FeltRepresentation::SingleHex(base_element) => Ok(base_element),
+            FeltRepresentation::SingleDecimal(base_element) => Ok(base_element),
+            FeltRepresentation::Dynamic(template_key) => template_values
+                .get(template_key.inner())
+                .ok_or_else(|| {
+                    AccountComponentTemplateError::TemplateValueNotProvided(
+                        template_key.inner().to_string(),
+                    )
+                })?
+                .as_felt()
+                .copied(),
+        }
+    }
+}
+
+impl Default for FeltRepresentation {
+    fn default() -> Self {
+        FeltRepresentation::SingleHex(Felt::default())
+    }
+}
+
+impl<'de> Deserialize<'de> for FeltRepresentation {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value = String::deserialize(deserializer)?;
+        if let Some(hex_str) = value.strip_prefix("0x").or_else(|| value.strip_prefix("0X")) {
+            let felt_value = u64::from_str_radix(hex_str, 16).map_err(serde::de::Error::custom)?;
+            Ok(FeltRepresentation::SingleHex(Felt::new(felt_value)))
+        } else if let Ok(decimal_value) = value.parse::<u64>() {
+            Ok(FeltRepresentation::SingleDecimal(Felt::new(decimal_value)))
```

Comment:
I don't think we would want to silently wrap around if the given `u64` exceeds the field modulus. If users specify such a value it would always be an error I would think, so it would be better to use `Felt::try_from` and propagate the error.

Reply:
Good catch, changed